storage/client_test.go

```diff
@@ -417,6 +417,32 @@ func TestListDefaultObjectACLsEmulated(t *testing.T) {
 	})
 }
 
+func TestUpdateBucketACLEmulated(t *testing.T) {
+	transportClientTest(t, func(t *testing.T, project, bucket string, client storageClient) {
+		ctx := context.Background()
+		attrs := &BucketAttrs{
+			Name: bucket,
+		}
+		// Create the bucket that will be retrieved.
+		if _, err := client.CreateBucket(ctx, project, attrs); err != nil {
+			t.Fatalf("client.CreateBucket: %v", err)
+		}
+		entity := AllUsers
+		role := RoleReader
+		want := &ACLRule{Entity: entity, Role: role}
+		got, err := client.UpdateBucketACL(ctx, bucket, entity, role)
```

Comment:
Could we write a test that also verifies that updating does not nuke existing rules? It would be interesting to use PredefinedACL in the bucket creation to get this, maybe.

Reply:
Good idea! Added PredefinedACL in the bucket creation and asserting with ListBucketACLs